src/core/thread/meshcop_dataset.cpp

```diff
@@ -239,6 +247,26 @@ ThreadError Dataset::Set(const otOperationalDataset &aDataset, bool aActive)
         Set(tlv);
     }
 
+    if (aDataset.mIsChannelMaskPage0Set)
+    {
+        MeshCoP::ChannelMaskTlv tlv;
+        MeshCoP::ChannelMaskEntry *entry;
+        uint8_t data[6];
+        tlv.Init();
+        tlv.SetLength(sizeof(MeshCoP::ChannelMaskEntry) + sizeof(aDataset.mChannelMaskPage0));
+
+        entry = reinterpret_cast<MeshCoP::ChannelMaskEntry *>(data);
+        entry->SetChannelPage(0);
+        entry->SetMaskLength(sizeof(aDataset.mChannelMaskPage0));
+
+        for (uint8_t index = 0; index < sizeof(aDataset.mChannelMaskPage0); index++)
+        {
+            data[index + sizeof(MeshCoP::ChannelMaskEntry)] = (aDataset.mChannelMaskPage0 >> (8 * index)) & 0xff;
+        }
+
+        Set(tlv, *data);
```

Comment:
I think we could simplify the above logic and avoid the need for defining this new `Set()` method by simply declaring a struct above.  For example:

```c
    if (aDataset.mIsChannelMaskPage0Set)
    {
        OT_TOOL_PACKED_BEGIN
        struct
        {
            MeshCoP::ChannelMaskTlv tlv;
            MeshCoP::ChannelMaskEntry entry;
            uint8_t mask[4];
        } OT_TOOL_PACKED_END channelMask;

        ...

        Set(channelMask.tlv);

        ...

    }
```